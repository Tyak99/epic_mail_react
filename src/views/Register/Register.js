import React, { Component, Fragment } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Link, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import style from './register.css';
import RegisterUi from '../../components/RegisterUi/RegisterUi';
import * as actionTypes from '../../../store/actions/auth';

export class Register extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: {},
    };
  }

  submitHandler = (e) => {
    e.preventDefault();
    const { register } = this.props;
    const {
      email,
      password,
      firstName,
      lastName,
      confirmPassword,
    } = this.state;
    const authData = {
      email,
      password,
      firstName,
      lastName,
      confirmPassword,
    };
    register(authData);
  }


  handleInput = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  }


  render() {
    let authRedirect;
    const { token, error, loading } = this.props;
    if (token) {
      authRedirect = <Redirect to="/inbox" />;
    }
    return (
      <Fragment>
        <div className={style.nav}>
          <Link to="/">
            <h2 className={style.navBrand}>
            EPIC MAIL
              <FontAwesomeIcon icon="mail-bulk" className={style.icon} />
            </h2>
          </Link>
        </div>
        <RegisterUi
          handleInput={this.handleInput}
          submit={e => this.submitHandler(e)}
          error={error}
          redirect={authRedirect}
          loading={loading}
        />
      </Fragment>
    );
  }
}

Register.defaultProps = {
  error: null,
  token: null,
  loading: false,
};
Register.propTypes = {
  register: PropTypes.func.isRequired,
  token: PropTypes.string,
  error: PropTypes.string,
  loading: PropTypes.bool,
};
const mapStateToprops = state => ({
  error: state.auth.error,
  token: state.auth.data.token,
  loading: state.auth.isLoading,
});
const mapDispatchToProps = dispatch => ({
  register: authData => dispatch(actionTypes.register(authData)),
});
export default connect(mapStateToprops, mapDispatchToProps)(Register);
