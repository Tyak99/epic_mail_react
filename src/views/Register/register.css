  .nav {
    overflow: hidden;
    background-color: #3C91E6;
  }
  
  .navBrand {
    float: left;
    margin: 12px 21px 13px 12px;
    color: white;
    text-transform: uppercase;
  }
  
  .nav a:first-child {
    padding: 0;
  }