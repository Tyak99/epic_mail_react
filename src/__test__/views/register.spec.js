import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { Register } from '../../views/Register/Register';

const props = {
  register: () => {},
  handleInput: () => {},
  submit: () => {},
};
describe('component: Register', () => {
  let RegisterWrapper;
  let handleInput;
  let submitHandler;
  beforeEach(() => {
    RegisterWrapper = shallow(<Register {...props} />);
    handleInput = sinon.spy(RegisterWrapper.instance(), 'handleInput');
    submitHandler = sinon.spy(RegisterWrapper.instance(), 'submitHandler');
    RegisterWrapper.instance().forceUpdate();
  });
  it('should render correctly', (done) => {
    const wrapper = shallow(<Register {...props} />);
    expect(wrapper).toMatchSnapshot();
    done();
  });
  it('should render correctly when token is passed along', (done) => {
    const wrapper = shallow(<Register {...props} token="hgahgs" />);
    expect(wrapper).toMatchSnapshot();
    done();
  });
  it('should call the input handler when email input changes', (done) => {
    const inputField = RegisterWrapper.find('RegisterUi').shallow().find('FormGroup').first()
      .shallow()
      .find('#firstName');
    const event = { target: { name: 'firstName', value: 'Tunde' } };
    inputField.simulate('change', event);
    expect(inputField.length).toBe(1);
    expect(RegisterWrapper.state().firstName).toEqual('Tunde');
    expect(handleInput.called).toBe(true);
    done();
  });
  it('should call the submit handler when submit is clicked', (done) => {
    const form = RegisterWrapper.find('RegisterUi').shallow().find('form');
    const event = { preventDefault: () => {} };
    form.simulate('submit', event);
    expect(form.length).toBe(1);
    expect(submitHandler.called).toBe(true);
    done();
  });
});
