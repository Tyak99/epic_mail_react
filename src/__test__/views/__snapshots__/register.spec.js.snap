// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Register should render correctly 1`] = `ShallowWrapper {}`;

exports[`component: Register should render correctly when token is passed along 1`] = `ShallowWrapper {}`;
